src/js_lexer.zig

```diff
@@ -273,6 +273,27 @@ fn NewLexer_(
             // }
         }
 
+        pub fn restore(this: *LexerType, original: *const LexerType) void {
```

Comment:
The length is not the main problem - the main problem is on array resize, the previous pointer is now invalid memory. This wouldn't fix it, as it reuses the memory. 

I think we would instead need to lazily clone it

Reply:
it copies the possibly new pointer from the new lexer, and sets the length from the original lexer. This way we can ensure the pointers are valid, new values aren't used, and memory isn't wasted.

doing this for `temp_buffer_u16` is unnecessary though, it's length is set to 0 after each use so we can ignore it. Will update

Reply:
the pointer can become invalid memory though when its resized later. even if the ptr and length match up.